multiuser/api/che-multiuser-api-authentication-commons/src/main/java/org/eclipse/che/multiuser/api/authentication/commons/token/HeaderRequestTokenExtractor.java

```diff
@@ -12,17 +12,25 @@
 package org.eclipse.che.multiuser.api.authentication.commons.token;
 
 import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.BadRequestException;
 import javax.ws.rs.core.HttpHeaders;
 
 /** Extract sso token from request headers. */
 public class HeaderRequestTokenExtractor implements RequestTokenExtractor {
+
   @Override
   public String getToken(HttpServletRequest req) {
     if (req.getHeader(HttpHeaders.AUTHORIZATION) == null) {
       return null;
     }
-    return req.getHeader(HttpHeaders.AUTHORIZATION).toLowerCase().startsWith("bearer")
-        ? req.getHeader(HttpHeaders.AUTHORIZATION).split(" ")[1]
-        : req.getHeader(HttpHeaders.AUTHORIZATION);
+    if (req.getHeader(HttpHeaders.AUTHORIZATION).toLowerCase().startsWith("bearer")) {
+      String[] parts = req.getHeader(HttpHeaders.AUTHORIZATION).split("\\s+");
+      if (parts.length != 2) {
+        throw new BadRequestException("Invalid authorization header format.");
```

Comment:
Ok for me, but it can contain sensitive data...

Reply:
yes, that's my concern as well. What is lifetime of a bearer token?

Reply:
The lifetime of a token is configurable in the IDP so we cannot assume anything here IMHO.

Reply:
that's unfortunate, because here we're fixing just consequence of some other possible issue. Even if it is improvement, it would be nice to have something that would help with diagnosing what is the cause of this.